var/spack/repos/builtin/packages/celeritas/package.py

```diff
@@ -125,6 +125,9 @@ def cmake_args(self):
         for pkg in ["CUDA", "Geant4", "HepMC3", "OpenMP", "ROOT", "SWIG", "VecGeom"]:
             args.append(from_variant("CELERITAS_USE_" + pkg, pkg.lower()))
 
+        if self.spec.satisfies("+cuda"):
+            args.append("CMAKE_CUDA_ARCHITECTURES", self.spec.variants["cuda_arch"].value)
```

Comment:
Wow, can't believe I missed that. Digging around to make sure it wasn't done in the cmake build system, we should probably do this (including adding the HIP args):
```suggestion
        if spec.satisfies("+cuda"):
            args.append(CMakeBuilder.define_cuda_architectures(self))
        if spec.satisfies("+rocm"):
            args.append(CMakeBuilder.define_hip_architectures(self))
```
with
```python
from spack.build_systems.cmake import CMakeBuilder
```

Reply:
(It looks like this form was made a little more awkward by #47576... and I feel like the resulting code 
should be added automatically to the build arguments.)